### PR TITLE
wlsunset: add module

### PR DESCRIPTION
**Pull request by matrss**
### Description
This PR adds a service module for `wlsunset`, a program for day/night gamma adjustments
on wayland. This module was requested in #1625.

Only think that I am not quite satisfied with is the description for `gamma`, but I couldn't figure out what exactly it does and upstream doesn't seem to say much about it aswell.

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Review by berbiche — APPROVED** [review]
LGTM!

Would you mind adding a simple test case like Kanshi's (`tests/modules/services/kanshi`)?

**Comment by matrss** [comment]
I wasn't really sure at first how to test this, because it basically just creates a service file. I took some inspiration from the tests for `sxhkd` where the service file is checked against some regex's, does this look like what you had in mind? It basically just compares the created service file for some arbitrary values against what is expected for those values.

**Comment by berbiche** [comment]
> I wasn't really sure at first how to test this, because it basically just creates a service file. I took some inspiration from the tests for `sxhkd` where the service file is checked against some regex's, does this look like what you had in mind? It basically just compares the created service file for some arbitrary values against what is expected for those values.

Tests usually use the function `assertFileContent` instead of regex. You can find an example of said function in Kanshi's test.

**Comment by matrss** [comment]
Yes, I have seen that, but `kanshi` does not actually check whats in the service file, just that it is created. It does use `assertFileContent` for it's config file, but `wlsunset` does not have a separate one. It's not quite clear to me how I would handle the absolute path in `ExecStart` in this case, can't really hardcode it. Should I do some "templating" as done in `tests/modules/emacs/emacs-service-emacs.service` and `tests/modules/services/lieer-service-expected.service`?

**Comment by berbiche** [comment]
One way to test the service file is to override the package:
`package = pkgs.writeScriptBin "wlsunset" "" // { outPath = "@wlsunset@"; };`
and do an `assertFileContent`.

We usually don't want to bring any external dependencies inside a home-manager test closure (so we define overlays or override the package).

Also, thanks for putting in the time in this contribution!

**Comment by matrss** [comment]
I think I figured it out now. Could you take a look again?

**Comment by berbiche** [comment]
One last thing I forgot to mention: you need to add an entry to `modules/misc/news.nix` when adding a new module.
Everything will be in order after and we will have to wait for rycee to merge.

**Comment by rycee** [comment]
@berbiche There is no need to add a news entry for a new module unless you want to have additional text over the usual "a new module was added: blah". See [3.2.9. Format your news entries](https://rycee.gitlab.io/home-manager/index.html#sec-guidelines-news-style). Thanks for reviewing so nicely, btw!

**Comment by rycee** [comment]
Thanks for the contribution! Merged to master.

**Comment by matrss** [comment]
Thanks for maintaining this awesome project! It was a pleasure to contribute, looking forward to the next time.

**Comment by bnjmnt4n** [comment]
Just wondering, was there any reason not to include a restart option in the service? I've had to manually start wlsunset a few times since there appear to be occasional issues with it starting up. If this was an oversight, I'd be willing to send a PR for this, similar to how it's done for [redshift](https://github.com/nix-community/home-manager/blob/115e76ae12a81bce5dcd19714cdeaaa8d5ca3ce8/modules/services/redshift.nix#L158-159).

**Comment by berbiche** [comment]
@bnjmnt4n In the meantime, you can manually set it to restart in your home configuration:

```nix
{
  systemd.user.services.wlsunset.Service = {
    Restart = "always";
    RestartSec = 3;
  };
}
```

**Comment by matrss** [comment]
I didn't include a restart option because I do not need it on my system and didn't think about it possibly being needed. I am not opposed to including one, although I am usually of the opinion that a restart on crash only fixes the symptom, but not the cause. Out of curiosity: does it only fail to start when starting the graphical session? If that's the case my wild guess would be a race condition between the compositor and `wlsunset`; what error message does `wlsunset` produce? The service might be missing a `After=` clause in that case, but I am no expert in systemd service file semantics...

**Pull request merged**
